### DISCUSSION TITLE
Problem increasing ef_construction with Milvus v2.3.2-dev

### DISCUSSION THREAD

**Discussion by wangyingdesign** [milvus-io/milvus#28855] [Q&A and General discussion]
Hi, I am trying to build index with the stored data. with efConstruction to be 1000, but it failed with error saying it can not exceed 512. I checked the milvus version is v2.3.2-dev, according to this doc https://milvus.io/docs/index.md this should work, could someone help with this? thank you!

**Comment by yhmo** [comment]
Good catch. It is a bug.
In v2.2.x, the range of M is [4, 64], range of efConstruction is [8, 512]
The search engine has changed the ranges to: M (1, 2048), efConstruction (1, int32_max). But in the upper layer there is a duplicate checker till using the old ranges to verify index parameters.

I will create an issue for this bug: https://github.com/milvus-io/milvus/issues/28860
Before we fix this bug, you have to follow the v2.2 doc to set index parameters: https://milvus.io/docs/v2.2.x/index.md#HNSW
